src/formatDate.js

```diff
@@ -8,7 +8,86 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let result = '';
+  const oldDate = date;
+  const newFormat = toFormat;
+  let array = [];
+  let year = '';
+  let yearBig = '';
+  const newArray = [];
+  let indexYear = 0;
+  let indexYearbig = 0;
+
+  const [, , , oldSeparator] = fromFormat;
+  const [, , , newSeparator] = newFormat;
+
+  array = oldDate.split(oldSeparator);
+
+  if (newFormat.includes('YYYY') && fromFormat.includes('YYYY')) {
+    for (const n of newFormat) {
+      const searchInd = fromFormat.indexOf(n);
+
+      newArray.push(array[searchInd]);
+    }
+    newArray.pop();
+    result = newArray.join(newSeparator);
+
+    return result;
+  }
+
+  if (newFormat.includes('YY')) {
+    for (const n of array) {
+      if (n.length > 3) {
+        indexYear = array.indexOf(n);
+        year = n.slice(2);
+      }
+    }
+    array[indexYear] = year;
+
+    const newYY = newFormat.indexOf('YY');
+
+    newFormat[newYY] = 'YYYY';
+
+    for (const n of newFormat) {
+      const searchInd = fromFormat.indexOf(n);
+
+      newArray.push(array[searchInd]);
+    }
+    newArray.pop();
+
+    result = newArray.join(newSeparator);
+
+    return result;
+  }
+
+  if (newFormat.includes('YYYY')) {
+    indexYearbig = fromFormat.indexOf('YY');
+
+    if (parseInt(array[indexYearbig]) >= 30) {
+      yearBig = '19' + array[indexYearbig];
+    }
+
+    if (parseInt(array[indexYearbig]) < 30) {
+      yearBig = '20' + array[indexYearbig];
+    }
+
+    array[indexYearbig] = yearBig;
+
+    const newY = newFormat.indexOf('YYYY');
+
+    newFormat[newY] = 'YY';
+
+    for (const n of newFormat) {
+      const searchInd = fromFormat.indexOf(n);
+
+      newArray.push(array[searchInd]);
+    }
+    newArray.pop();
+
+    result = newArray.join(newSeparator);
+
+    return result;
+  }
```

Comment:
Similar to the previous block, this section modifies `newFormat` directly. It's better to work with a copy to avoid side effects. Additionally, the logic for determining the century based on 'YY' is hardcoded to 30, which might not be flexible for all use cases.